homeassistant/components/media_player/directv.py

```diff
@@ -360,7 +360,11 @@ def media_next_track(self):
         _LOGGER.debug("Fast forward on %s", self._name)
         self.dtv.key_press('ffwd')
 
-    def select_source(self, source):
+    def play_media(self, media_type, media_id, **kwargs):
         """Select input source."""
-        _LOGGER.debug("Changing channel on %s to %s", self._name, source)
-        self.dtv.tune_channel(source)
+        if media_type != MEDIA_TYPE_CHANNEL:
+            _LOGGER.error("Invalid media type %s. Only %s is supported",
+                          media_type, MEDIA_TYPE_CHANNEL)
+        else:
+            _LOGGER.debug("Changing channel on %s to %s", self._name, media_id)
+            self.dtv.tune_channel(media_id)
```

Comment:
It's generally not good form to put the successful case in an `else`. Try a guard clause:

```python
def play_media(self, media_type, media_id, **kwargs):
    """Select input source."""
    if media_type != MEDIA_TYPE_CHANNEL:
        _LOGGER.error("Invalid media type %s. Only %s is supported",
                      media_type, MEDIA_TYPE_CHANNEL)
        return

    _LOGGER.debug("Changing channel on %s to %s", self._name, media_id)
    self.dtv.tune_channel(media_id)
```

Reply:
@bachya, I'm assuming when you say "guard clause" you mean what you have in your example. :-)

Updated accordingly and pushed.